agents/aws-to-gcp/src/main/scala/org/broadinstitute/transporter/transfer/AwsToGcpRunner.scala

```diff
@@ -35,32 +36,85 @@ class AwsToGcpRunner(
 ) extends TransferRunner[AwsToGcpRequest, AwsToGcpProgress, AwsToGcpOutput] {
   import AwsToGcpRunner._
 
-  override def initialize(request: AwsToGcpRequest): AwsToGcpProgress = {
+  override def initialize(
+    request: AwsToGcpRequest
+  ): Either[AwsToGcpProgress, AwsToGcpOutput] = {
+
+    val forceTransfer = request.force.getOrElse(false)
 
     val initFlow = for {
+
+      // First, check if the source S3 object exists and is readable.
       s3Metadata <- getS3Metadata(
         bucket = request.s3Bucket,
         region = request.s3Region,
         path = request.s3Path,
         expectedSize = request.expectedSize
       )
-      uploadToken <- initGcsResumableUpload(
+
+      /*
+       * Second, check if there's already an object at the GCS target.
+       *
+       * If there is an existing object, and it has a registered md5 hash,
+       * check if the md5 matches our expected final state. We use the
+       * comparison to provide better UX for repeated transfers of the
+       * same file.
+       */
+      (gcsExists, existingMd5) <- checkExistingObject(
         bucket = request.gcsBucket,
-        path = request.gcsPath,
-        s3Metadata = s3Metadata,
-        expectedMd5 = request.expectedMd5
+        path = request.gcsPath
       )
+      md5sMatch = request.expectedMd5.isDefined && existingMd5 == request.expectedMd5
```

Comment:
Maybe a comment or for readability explain/add checking that expected is defined vs checking that the existing is defined vs checking both